Adding admin-scp feature

dbadola opened the pull request:
adding admin-scp variable in "config system global" section. This help to enable config file upload

resource "fortios_system_setting_global" "fw1" {
  hostname     = "fw1"
  admintimeout = 460
  admin_scp = "enable"
}

davewongillies commented on vendor/github.com/fgtdev/fortios-sdk-go/sdkcore/system_setting_global.go:
```diff
@@ -16,6 +16,7 @@ type JSONSystemSettingGlobal struct {
 	Hostname     string `json:"hostname"`
 	AdminSport   string `json:"admin-sport"`
 	AdminSSHPort string `json:"admin-ssh-port"`
+	AdminScp     string `json:"admin-scp"`
```
You really should be raising a PR against https://github.com/fgtdev/fortios-sdk-go, get that merged and tagged, _then_ vendor the changes to here for this PR.

dbadola replied:
@davewongillies  Apologies, have created the same https://github.com/fgtdev/fortios-sdk-go/pull/2

frankshen01 commented:
Hi, @dbadola, great,  thank you for this. Looks good to me! But could you update the fortios_system_setting_global.html.markdown and resource_system_setting_global_test.go files as well?

dbadola commented:
Hi @frankshen01, I have updated the " resource_system_setting_global_test.go"  Where i can find "fortios_system_setting_global.html.markdown" ? Can you help me to point out the location?

dbadola commented:
@frankshen01 , I am trying to run the `make testacc` also tried to run `./test/test.sh` but i am getting below  FORTIOS_INSECURE error. What should I need to do here ?

```
=== RUN   TestProvider
--- PASS: TestProvider (0.00s)
=== RUN   TestAccFortiOSFirewallObjectAddress_withIPRange
--- FAIL: TestAccFortiOSFirewallObjectAddress_withIPRange (0.00s)
    provider_test.go:35: FORTIOS_INSECURE must be set for acceptance tests
=== RUN   TestAccFortiOSFirewallObjectAddress_withGeography
--- FAIL: TestAccFortiOSFirewallObjectAddress_withGeography (0.00s)
    provider_test.go:35: FORTIOS_INSECURE must be set for acceptance tests
=== RUN   TestAccFortiOSFirewallObjectAddress_withFqdn
--- FAIL: TestAccFortiOSFirewallObjectAddress_withFqdn (0.00s)
    provider_test.go:35: FORTIOS_INSECURE must be set for acceptance tests
=== RUN   TestAccFortiOSFirewallObjectAddressGroup_basic
--- FAIL: TestAccFortiOSFirewallObjectAddressGroup_basic (0.00s)
    provider_test.go:35: FORTIOS_INSECURE must be set for acceptance tests
=== RUN   TestAccFortiOSFirewallObjectIPPool_Overload
--- FAIL: TestAccFortiOSFirewallObjectIPPool_Overload (0.00s)
    provider_test.go:35: FORTIOS_INSECURE must be set for acceptance tests
=== RUN   TestAccFortiOSFirewallObjectIPPool_OneToOne
--- FAIL: TestAccFortiOSFirewallObjectIPPool_OneToOne (0.00s)
    provider_test.go:35: FORTIOS_INSECURE must be set for acceptance tests
=== RUN   TestAccFortiOSFirewallObjectService_Fqdn
--- FAIL: TestAccFortiOSFirewallObjectService_Fqdn (0.00s)
    provider_test.go:35: FORTIOS_INSECURE must be set for acceptance tests
=== RUN   TestAccFortiOSFirewallObjectService_IPRange
--- FAIL: TestAccFortiOSFirewallObjectService_IPRange (0.00s)
    provider_test.go:35: FORTIOS_INSECURE must be set for acceptance tests
=== RUN   TestAccFortiOSFirewallObjectServiceGroup_basic
--- FAIL: TestAccFortiOSFirewallObjectServiceGroup_basic (0.00s)
    provider_test.go:35: FORTIOS_INSECURE must be set for acceptance tests
=== RUN   TestAccFortiOSFirewallObjectVip_basic
--- FAIL: TestAccFortiOSFirewallObjectVip_basic (0.00s)
    provider_test.go:35: FORTIOS_INSECURE must be set for acceptance tests
=== RUN   TestAccFortiOSFirewallObjectVipGroup_basic
--- FAIL: TestAccFortiOSFirewallObjectVipGroup_basic (0.00s)
    provider_test.go:35: FORTIOS_INSECURE must be set for acceptance tests
=== RUN   TestAccFortiOSFirewallSecurityPolicy_basic
--- FAIL: TestAccFortiOSFirewallSecurityPolicy_basic (0.00s)
    provider_test.go:35: FORTIOS_INSECURE must be set for acceptance tests
=== RUN   TestAccFortiOSLogFortiAnalyzerSetting_basic
--- FAIL: TestAccFortiOSLogFortiAnalyzerSetting_basic (0.00s)
    provider_test.go:35: FORTIOS_INSECURE must be set for acceptance tests
=== RUN   TestAccFortiOSLogSyslogSetting_basic
--- FAIL: TestAccFortiOSLogSyslogSetting_basic (0.00s)
    provider_test.go:35: FORTIOS_INSECURE must be set for acceptance tests
=== RUN   TestAccFortiOSNetworkingInterfacePort_Physical
--- FAIL: TestAccFortiOSNetworkingInterfacePort_Physical (0.00s)
    provider_test.go:35: FORTIOS_INSECURE must be set for acceptance tests
=== RUN   TestAccFortiOSNetworkingInterfacePort_Vlan
--- FAIL: TestAccFortiOSNetworkingInterfacePort_Vlan (0.00s)
    provider_test.go:35: FORTIOS_INSECURE must be set for acceptance tests
=== RUN   TestAccFortiOSNetworkingInterfacePort_Loopback
--- FAIL: TestAccFortiOSNetworkingInterfacePort_Loopback (0.00s)
    provider_test.go:35: FORTIOS_INSECURE must be set for acceptance tests
=== RUN   TestAccFortiOSNetworkingRouteStatic_basic
--- FAIL: TestAccFortiOSNetworkingRouteStatic_basic (0.00s)
    provider_test.go:35: FORTIOS_INSECURE must be set for acceptance tests
=== RUN   TestAccFortiOSSystemAdminAdministrator_basic
--- FAIL: TestAccFortiOSSystemAdminAdministrator_basic (0.00s)
    provider_test.go:35: FORTIOS_INSECURE must be set for acceptance tests
=== RUN   TestAccFortiOSSystemAdminProfiles_basic
--- FAIL: TestAccFortiOSSystemAdminProfiles_basic (0.00s)
    provider_test.go:35: FORTIOS_INSECURE must be set for acceptance tests
=== RUN   TestAccFortiOSSystemAPIUserSetting_basic
--- FAIL: TestAccFortiOSSystemAPIUserSetting_basic (0.00s)
    provider_test.go:35: FORTIOS_INSECURE must be set for acceptance tests
=== RUN   TestAccFortiOSSystemSettingDNS_basic
--- FAIL: TestAccFortiOSSystemSettingDNS_basic (0.00s)
    provider_test.go:35: FORTIOS_INSECURE must be set for acceptance tests
=== RUN   TestAccFortiOSSystemSettingGlobal_basic
--- FAIL: TestAccFortiOSSystemSettingGlobal_basic (0.00s)
    provider_test.go:35: FORTIOS_INSECURE must be set for acceptance tests
=== RUN   TestAccFortiOSSystemSettingNTP_basic
--- FAIL: TestAccFortiOSSystemSettingNTP_basic (0.00s)
    provider_test.go:35: FORTIOS_INSECURE must be set for acceptance tests
=== RUN   TestAccFortiOSVPNIPsecPhase1Interface_basic
--- FAIL: TestAccFortiOSVPNIPsecPhase1Interface_basic (0.00s)
    provider_test.go:35: FORTIOS_INSECURE must be set for acceptance tests
=== RUN   TestAccFortiOSVPNIPsecPhase1Interface_basic2
--- FAIL: TestAccFortiOSVPNIPsecPhase1Interface_basic2 (0.00s)
    provider_test.go:35: FORTIOS_INSECURE must be set for acceptance tests
=== RUN   TestAccFortiOSVPNIPsecPhase2Interface_basic1
--- FAIL: TestAccFortiOSVPNIPsecPhase2Interface_basic1 (0.00s)
    provider_test.go:35: FORTIOS_INSECURE must be set for acceptance tests
=== RUN   TestAccFortiOSVPNIPsecPhase2Interface_basic2
--- FAIL: TestAccFortiOSVPNIPsecPhase2Interface_basic2 (0.00s)
    provider_test.go:35: FORTIOS_INSECURE must be set for acceptance tests
FAIL
FAIL    github.com/terraform-providers/terraform-provider-fortios/fortios       0.073s
make: *** [testacc] Error 1
```

frankshen01 commented:
> Hi @frankshen01, I have updated the " resource_system_setting_global_test.go" Where i can find "fortios_system_setting_global.html.markdown" ? Can you help me to point out the location?

Hi @dbadola, thanks for your extra effort. Please check:  https://github.com/terraform-providers/terraform-provider-fortios/blob/master/website/docs/r/fortios_system_setting_global.html.markdown

frankshen01 commented:
> @frankshen01 , I am trying to run the `make testacc` also tried to run `./test/test.sh` but i am getting below FORTIOS_INSECURE error. What should I need to do here ?
> 
> ```
> === RUN   TestProvider
> --- PASS: TestProvider (0.00s)
> === RUN   TestAccFortiOSFirewallObjectAddress_withIPRange
> --- FAIL: TestAccFortiOSFirewallObjectAddress_withIPRange (0.00s)
> ........
>     provider_test.go:35: FORTIOS_INSECURE must be set for acceptance tests
> === RUN   TestAccFortiOSVPNIPsecPhase2Interface_basic2
> --- FAIL: TestAccFortiOSVPNIPsecPhase2Interface_basic2 (0.00s)
>     provider_test.go:35: FORTIOS_INSECURE must be set for acceptance tests
> FAIL
> FAIL    github.com/terraform-providers/terraform-provider-fortios/fortios       0.073s
> make: *** [testacc] Error 1
> ```

Hi @dbadola , new environment variables have been added, you can set FORTIOS_INSECURE to "true", keep FORTIOS_CA_CABUNDLE empty, then it should solve the problem. Please check the latest test.sh for reference. 

For details on these two parameters, please refer to: Part "Argument Reference" of https://github.com/terraform-providers/terraform-provider-fortios/blob/master/website/docs/index.html.markdown

dbadola commented:
@frankshen01, Thank you for providing me the details.I have updated the documentation markup file

> > Hi @frankshen01, I have updated the " resource_system_setting_global_test.go" Where i can find "fortios_system_setting_global.html.markdown" ? Can you help me to point out the location?
> 
> Hi @dbadola, thanks for your extra effort. Please check: https://github.com/terraform-providers/terraform-provider-fortios/blob/master/website/docs/r/fortios_system_setting_global.html.markdown

dbadola commented:
@frankshen01,  I am getting test fails in different section (vpn,address, policy) in the base (master) repo itself even though my changes are not related to it.  Is the test passing for you ?

```
?       github.com/terraform-providers/terraform-provider-fortios       [no test files]
=== RUN   TestProvider
--- PASS: TestProvider (0.00s)
=== RUN   TestAccFortiOSFirewallObjectAddress_withIPRange
--- FAIL: TestAccFortiOSFirewallObjectAddress_withIPRange (2.67s)
    testing.go:568: Step 0 error: After applying this step, the plan was not empty:
        
        DIFF:
        
        UPDATE: fortios_firewall_object_address.test1
          comment:  "Terraform Test" => "Terraform Test"
          country:  "\x01" => ""
          end_ip:   "2.0.0.0" => "2.0.0.0"
          fqdn:     "" => ""
          id:       "foair_zbdknvqxvm0m" => "foair_zbdknvqxvm0m"
          name:     "foair_zbdknvqxvm0m" => "foair_zbdknvqxvm0m"
          start_ip: "1.0.0.0" => "1.0.0.0"
          subnet:   "1.0.0.0 2.0.0.0" => "1.0.0.0 2.0.0.0"
          type:     "iprange" => "iprange"
        
        
        
        STATE:
        
        fortios_firewall_object_address.test1:
          ID = foair_zbdknvqxvm0m
          provider = provider.fortios
          comment = Terraform Test
          country = 
          end_ip = 2.0.0.0
          fqdn = 
          name = foair_zbdknvqxvm0m
          start_ip = 1.0.0.0
          subnet = 1.0.0.0 2.0.0.0
          type = iprange
=== RUN   TestAccFortiOSFirewallObjectAddress_withGeography
--- PASS: TestAccFortiOSFirewallObjectAddress_withGeography (3.39s)
=== RUN   TestAccFortiOSFirewallObjectAddress_withFqdn
--- PASS: TestAccFortiOSFirewallObjectAddress_withFqdn (3.37s)
=== RUN   TestAccFortiOSFirewallObjectAddressGroup_basic
--- PASS: TestAccFortiOSFirewallObjectAddressGroup_basic (3.85s)
=== RUN   TestAccFortiOSFirewallObjectIPPool_Overload
--- PASS: TestAccFortiOSFirewallObjectIPPool_Overload (3.39s)
=== RUN   TestAccFortiOSFirewallObjectIPPool_OneToOne
--- PASS: TestAccFortiOSFirewallObjectIPPool_OneToOne (3.46s)
=== RUN   TestAccFortiOSFirewallObjectService_Fqdn
--- PASS: TestAccFortiOSFirewallObjectService_Fqdn (3.41s)
=== RUN   TestAccFortiOSFirewallObjectService_IPRange
--- PASS: TestAccFortiOSFirewallObjectService_IPRange (3.40s)
=== RUN   TestAccFortiOSFirewallObjectServiceGroup_basic
--- PASS: TestAccFortiOSFirewallObjectServiceGroup_basic (3.37s)
=== RUN   TestAccFortiOSFirewallObjectVip_basic
--- PASS: TestAccFortiOSFirewallObjectVip_basic (3.40s)
=== RUN   TestAccFortiOSFirewallObjectVipGroup_basic
--- PASS: TestAccFortiOSFirewallObjectVipGroup_basic (4.63s)
=== RUN   TestAccFortiOSFirewallSecurityPolicy_basic
--- FAIL: TestAccFortiOSFirewallSecurityPolicy_basic (0.74s)
    testing.go:568: Step 0 error: errors during apply:
        
        Error: Error creating Firewall Security Policy: status is error and error no is -3 and http_status no is 500
        
          on /tmp/tf-test672029927/main.tf line 2:
          (source code not available)
        
        
=== RUN   TestAccFortiOSLogFortiAnalyzerSetting_basic
--- PASS: TestAccFortiOSLogFortiAnalyzerSetting_basic (3.41s)
=== RUN   TestAccFortiOSLogSyslogSetting_basic
--- FAIL: TestAccFortiOSLogSyslogSetting_basic (2.51s)
    testing.go:568: Step 0 error: Check failed: Check 4/8 error: fortios_log_syslog_setting.test1: Attribute 'mode' expected "udp", got ""
=== RUN   TestAccFortiOSNetworkingInterfacePort_Physical
--- PASS: TestAccFortiOSNetworkingInterfacePort_Physical (2.70s)
=== RUN   TestAccFortiOSNetworkingInterfacePort_Vlan
--- PASS: TestAccFortiOSNetworkingInterfacePort_Vlan (4.04s)
=== RUN   TestAccFortiOSNetworkingInterfacePort_Loopback
--- FAIL: TestAccFortiOSNetworkingInterfacePort_Loopback (3.40s)
    testing.go:568: Step 0 error: Check failed: Check 9/12 error: fortios_networking_interface_port.test3: Attribute 'role' expected "lan", got "undefined"
=== RUN   TestAccFortiOSNetworkingRouteStatic_basic
--- PASS: TestAccFortiOSNetworkingRouteStatic_basic (3.38s)
=== RUN   TestAccFortiOSSystemAdminAdministrator_basic
--- FAIL: TestAccFortiOSSystemAdminAdministrator_basic (3.40s)
    testing.go:568: Step 0 error: After applying this step, the plan was not empty:
        
        DIFF:
        
        UPDATE: fortios_system_admin_profiles.test1
          admintimeout_override: "disable" => "disable"
          authgrp:               "none" => "none"
          comments:              "Terraform Test" => "Terraform Test"
          ftviewgrp:             "" => "read"
          fwgrp:                 "none" => "none"
          id:                    "profilemdmn7ceu" => "profilemdmn7ceu"
          loggrp:                "none" => "none"
          name:                  "profilemdmn7ceu" => "profilemdmn7ceu"
          netgrp:                "none" => "none"
          scope:                 "vdom" => "vdom"
          secfabgrp:             "" => "none"
          sysgrp:                "read" => "read"
          utmgrp:                "none" => "none"
          vpngrp:                "none" => "none"
          wanoptgrp:             "none" => "none"
          wifi:                  "none" => "none"
        
        
        
        STATE:
        
        fortios_system_admin_administrator.test1:
          ID = mdmn7ceu
          provider = provider.fortios
          accprofile = profilemdmn7ceu
          comments = Terraform Test
          name = mdmn7ceu
          password = cc37$1AC1
          trusthost1 = 1.1.1.0 255.255.255.0
          trusthost10 = 0.0.0.0 0.0.0.0
          trusthost2 = 2.2.2.0 255.255.255.0
          trusthost3 = 0.0.0.0 0.0.0.0
          trusthost4 = 0.0.0.0 0.0.0.0
          trusthost5 = 0.0.0.0 0.0.0.0
          trusthost6 = 0.0.0.0 0.0.0.0
          trusthost7 = 0.0.0.0 0.0.0.0
          trusthost8 = 0.0.0.0 0.0.0.0
          trusthost9 = 0.0.0.0 0.0.0.0
          vdom.# = 1
          vdom.0 = root
        
          Dependencies:
            fortios_system_admin_profiles.test1
        fortios_system_admin_profiles.test1:
          ID = profilemdmn7ceu
          provider = provider.fortios
          admintimeout_override = disable
          authgrp = none
          comments = Terraform Test
          ftviewgrp = 
          fwgrp = none
          loggrp = none
          name = profilemdmn7ceu
          netgrp = none
          scope = vdom
          secfabgrp = 
          sysgrp = read
          utmgrp = none
          vpngrp = none
          wanoptgrp = none
          wifi = none
=== RUN   TestAccFortiOSSystemAdminProfiles_basic
--- FAIL: TestAccFortiOSSystemAdminProfiles_basic (2.66s)
    testing.go:568: Step 0 error: Check failed: Check 5/16 error: fortios_system_admin_profiles.test1: Attribute 'secfabgrp' expected "none", got ""
=== RUN   TestAccFortiOSSystemAPIUserSetting_basic
--- FAIL: TestAccFortiOSSystemAPIUserSetting_basic (2.49s)
    testing.go:568: Step 0 error: errors during apply:
        
        Error: Error creating System APIUser Setting: status is error and error no is -37 and http_status no is 500
        
          on /tmp/tf-test971502657/main.tf line 20:
          (source code not available)
        
        
=== RUN   TestAccFortiOSSystemSettingDNS_basic
--- PASS: TestAccFortiOSSystemSettingDNS_basic (2.51s)
=== RUN   TestAccFortiOSSystemSettingGlobal_basic
--- PASS: TestAccFortiOSSystemSettingGlobal_basic (2.52s)
=== RUN   TestAccFortiOSSystemSettingNTP_basic
--- PASS: TestAccFortiOSSystemSettingNTP_basic (2.50s)
=== RUN   TestAccFortiOSVPNIPsecPhase1Interface_basic
--- PASS: TestAccFortiOSVPNIPsecPhase1Interface_basic (3.38s)
=== RUN   TestAccFortiOSVPNIPsecPhase1Interface_basic2
--- PASS: TestAccFortiOSVPNIPsecPhase1Interface_basic2 (3.25s)
=== RUN   TestAccFortiOSVPNIPsecPhase2Interface_basic1
--- FAIL: TestAccFortiOSVPNIPsecPhase2Interface_basic1 (3.69s)
    testing.go:568: Step 0 error: Check failed: Check 4/9 error: fortios_vpn_ipsec_phase2interface.test1: Attribute 'proposal' expected "aes128-sha1 aes256-sha1 aes128-sha256 aes256-sha256 aes128gcm aes256gcm chacha20poly1305", got "aes128-sha1 aes256-sha1 aes128-sha256 aes256-sha256 aes128gcm aes256gcm"
=== RUN   TestAccFortiOSVPNIPsecPhase2Interface_basic2
--- FAIL: TestAccFortiOSVPNIPsecPhase2Interface_basic2 (3.49s)
    testing.go:568: Step 0 error: Check failed: Check 4/13 error: fortios_vpn_ipsec_phase2interface.test2: Attribute 'proposal' expected "aes128-sha1 aes256-sha1 aes128-sha256 aes256-sha256 aes128gcm aes256gcm chacha20poly1305", got "aes128-sha1 aes256-sha1 aes128-sha256 aes256-sha256 aes128gcm aes256gcm"
FAIL
FAIL    github.com/terraform-providers/terraform-provider-fortios/fortios       88.674s
make: *** [testacc] Error 1
```

frankshen01 commented:
> @frankshen01, Thank you for providing me the details.I have updated the documentation markup file
> 
> > > Hi @frankshen01, I have updated the " resource_system_setting_global_test.go" Where i can find "fortios_system_setting_global.html.markdown" ? Can you help me to point out the location?
> > 
> > 
> > Hi @dbadola, thanks for your extra effort. Please check: https://github.com/terraform-providers/terraform-provider-fortios/blob/master/website/docs/r/fortios_system_setting_global.html.markdown

Hi @dbadola , thank you, we'll review and test it , if everything is ok, it will be merged into the next release.

frankshen01 commented:
> @frankshen01, I am getting test fails in different section (vpn,address, policy) in the base (master) repo itself even though my changes are not related to it. Is the test passing for you ?
> 
> ```
> ?       github.com/terraform-providers/terraform-provider-fortios       [no test files]
> === RUN   TestProvider
> --- PASS: TestProvider (0.00s)
> === RUN   TestAccFortiOSFirewallObjectAddress_withIPRange
> --- FAIL: TestAccFortiOSFirewallObjectAddress_withIPRange (2.67s)
>     testing.go:568: Step 0 error: After applying this step, the plan was not empty:
>         
>         DIFF:
>         
>         UPDATE: fortios_firewall_object_address.test1
>           comment:  "Terraform Test" => "Terraform Test"
>           country:  "\x01" => ""
>           end_ip:   "2.0.0.0" => "2.0.0.0"
>           fqdn:     "" => ""
>           id:       "foair_zbdknvqxvm0m" => "foair_zbdknvqxvm0m"
>           name:     "foair_zbdknvqxvm0m" => "foair_zbdknvqxvm0m"
>           start_ip: "1.0.0.0" => "1.0.0.0"
>           subnet:   "1.0.0.0 2.0.0.0" => "1.0.0.0 2.0.0.0"
>           type:     "iprange" => "iprange"
>         
>         
>         
> ......
> === RUN   TestAccFortiOSVPNIPsecPhase2Interface_basic1
> --- FAIL: TestAccFortiOSVPNIPsecPhase2Interface_basic1 (3.69s)
>     testing.go:568: Step 0 error: Check failed: Check 4/9 error: fortios_vpn_ipsec_phase2interface.test1: Attribute 'proposal' expected "aes128-sha1 aes256-sha1 aes128-sha256 aes256-sha256 aes128gcm aes256gcm chacha20poly1305", got "aes128-sha1 aes256-sha1 aes128-sha256 aes256-sha256 aes128gcm aes256gcm"
> === RUN   TestAccFortiOSVPNIPsecPhase2Interface_basic2
> --- FAIL: TestAccFortiOSVPNIPsecPhase2Interface_basic2 (3.49s)
>     testing.go:568: Step 0 error: Check failed: Check 4/13 error: fortios_vpn_ipsec_phase2interface.test2: Attribute 'proposal' expected "aes128-sha1 aes256-sha1 aes128-sha256 aes256-sha256 aes128gcm aes256gcm chacha20poly1305", got "aes128-sha1 aes256-sha1 aes128-sha256 aes256-sha256 aes128gcm aes256gcm"
> FAIL
> FAIL    github.com/terraform-providers/terraform-provider-fortios/fortios       88.674s
> make: *** [testacc] Error 1
> ```
@dbadola We have run this test many times and everything is OK. Please check if your device's FortiOS version is 6.0 or 6.2. Please check if your device relevant resources or database are updated to the latest status.

dbadola commented:
@frankshen01  so I tried the test again with 6.0.2 but it's failing do we have any dependency with these on `adminscp` ? 

TestAccFortiOSFirewallSecurityPolicy_basic, TestAccFortiOSSystemAPIUserSetting_basic, TestAccFortiOSSystemAPIUserSetting_basic


```
=== RUN   TestProvider
--- PASS: TestProvider (0.00s)
=== RUN   TestAccFortiOSFirewallObjectAddress_withIPRange
--- PASS: TestAccFortiOSFirewallObjectAddress_withIPRange (3.58s)
=== RUN   TestAccFortiOSFirewallObjectAddress_withGeography
--- PASS: TestAccFortiOSFirewallObjectAddress_withGeography (3.45s)
=== RUN   TestAccFortiOSFirewallObjectAddress_withFqdn
--- PASS: TestAccFortiOSFirewallObjectAddress_withFqdn (3.46s)
=== RUN   TestAccFortiOSFirewallObjectAddressGroup_basic
--- PASS: TestAccFortiOSFirewallObjectAddressGroup_basic (5.55s)
=== RUN   TestAccFortiOSFirewallObjectIPPool_Overload
--- PASS: TestAccFortiOSFirewallObjectIPPool_Overload (3.75s)
=== RUN   TestAccFortiOSFirewallObjectIPPool_OneToOne
--- PASS: TestAccFortiOSFirewallObjectIPPool_OneToOne (3.38s)
=== RUN   TestAccFortiOSFirewallObjectService_Fqdn
--- PASS: TestAccFortiOSFirewallObjectService_Fqdn (3.40s)
=== RUN   TestAccFortiOSFirewallObjectService_IPRange
--- PASS: TestAccFortiOSFirewallObjectService_IPRange (3.40s)
=== RUN   TestAccFortiOSFirewallObjectServiceGroup_basic
--- PASS: TestAccFortiOSFirewallObjectServiceGroup_basic (3.40s)
=== RUN   TestAccFortiOSFirewallObjectVip_basic
--- PASS: TestAccFortiOSFirewallObjectVip_basic (3.40s)
=== RUN   TestAccFortiOSFirewallObjectVipGroup_basic
--- PASS: TestAccFortiOSFirewallObjectVipGroup_basic (4.55s)
=== RUN   TestAccFortiOSFirewallSecurityPolicy_basic
--- FAIL: TestAccFortiOSFirewallSecurityPolicy_basic (1.15s)
    testing.go:568: Step 0 error: errors during apply:
        
        Error: Error creating Firewall Security Policy: status is error and error no is -3 and http_status no is 500
        
          on /var/folders/0c/dhwls6kn3rqf45q9__hcp1kr1jp3kg/T/tf-test253077806/main.tf line 2:
          (source code not available)
        
        
=== RUN   TestAccFortiOSLogFortiAnalyzerSetting_basic
--- PASS: TestAccFortiOSLogFortiAnalyzerSetting_basic (4.86s)
=== RUN   TestAccFortiOSLogSyslogSetting_basic
--- PASS: TestAccFortiOSLogSyslogSetting_basic (3.89s)
=== RUN   TestAccFortiOSNetworkingInterfacePort_Physical
--- PASS: TestAccFortiOSNetworkingInterfacePort_Physical (2.96s)
=== RUN   TestAccFortiOSNetworkingInterfacePort_Vlan
--- PASS: TestAccFortiOSNetworkingInterfacePort_Vlan (3.52s)
=== RUN   TestAccFortiOSNetworkingInterfacePort_Loopback
--- PASS: TestAccFortiOSNetworkingInterfacePort_Loopback (3.52s)
=== RUN   TestAccFortiOSNetworkingRouteStatic_basic
--- PASS: TestAccFortiOSNetworkingRouteStatic_basic (3.39s)
=== RUN   TestAccFortiOSSystemAdminAdministrator_basic
--- PASS: TestAccFortiOSSystemAdminAdministrator_basic (4.34s)
=== RUN   TestAccFortiOSSystemAdminProfiles_basic
--- PASS: TestAccFortiOSSystemAdminProfiles_basic (3.37s)
=== RUN   TestAccFortiOSSystemAPIUserSetting_basic
--- FAIL: TestAccFortiOSSystemAPIUserSetting_basic (2.52s)
    testing.go:568: Step 0 error: errors during apply:
        
        Error: Error creating System APIUser Setting: status is error and error no is -37 and http_status no is 500
        
          on /var/folders/0c/dhwls6kn3rqf45q9__hcp1kr1jp3kg/T/tf-test476715744/main.tf line 20:
          (source code not available)
        
        
=== RUN   TestAccFortiOSSystemSettingDNS_basic
--- PASS: TestAccFortiOSSystemSettingDNS_basic (2.51s)
=== RUN   TestAccFortiOSSystemSettingGlobal_basic
--- PASS: TestAccFortiOSSystemSettingGlobal_basic (2.51s)
=== RUN   TestAccFortiOSSystemSettingNTP_basic
--- PASS: TestAccFortiOSSystemSettingNTP_basic (2.51s)
=== RUN   TestAccFortiOSVPNIPsecPhase1Interface_basic
--- PASS: TestAccFortiOSVPNIPsecPhase1Interface_basic (3.40s)
=== RUN   TestAccFortiOSVPNIPsecPhase1Interface_basic2
--- PASS: TestAccFortiOSVPNIPsecPhase1Interface_basic2 (5.64s)
=== RUN   TestAccFortiOSVPNIPsecPhase2Interface_basic1
--- PASS: TestAccFortiOSVPNIPsecPhase2Interface_basic1 (4.30s)
=== RUN   TestAccFortiOSVPNIPsecPhase2Interface_basic2
--- PASS: TestAccFortiOSVPNIPsecPhase2Interface_basic2 (4.35s)
FAIL
FAIL    github.com/terraform-providers/terraform-provider-fortios/fortios       100.132s
make: *** [testacc] Error 1
```

frankshen01 commented:
hi, @dbadola, thanks for your hard work,  it seems that there is something wrong with the resource configuration. I will merge your code into the local  repository, and test to see what went wrong.

frankshen01 commented:
@dbadola , The test results on my side show that TestAccFortiOSSystemAPIUserSetting_basic is OK on both 6.0 and 6.2. TestAccFortiOSFirewallSecurityPolicy_basic is OK on 6.2, but there are some problems on 6.0. The reason is the lack of the Internet Service Database table entries, they need to be imported. I have changed the acc test file to avoid this trouble. It should be OK now. 

About TestAccFortiOSSystemAPIUserSetting_basic, please check if the following configuration can be satisfied on your device(By the way, you need to make the  APIUser to which the token belongs has the super_admin privilege, Otherwise it will have no permission to create other APIUsers)
```

resource "fortios_system_admin_profiles" "test1" {
	name = "apsbp"
	scope = "vdom"
	comments = "Terraform Test"
	secfabgrp = "none"
	ftviewgrp = "read"
	authgrp = "none"
	sysgrp = "read"
	netgrp = "none"
	loggrp = "none"
	fwgrp = "none"
	vpngrp = "none"
	utmgrp = "none"
	wanoptgrp = "none"
	wifi = "none"
	admintimeout_override = "disable"
}

resource "fortios_system_apiuser_setting" "test1" {
	name = "apiuser"
	accprofile = "${fortios_system_admin_profiles.test1.name}"
	vdom = ["root"]
	trusthost {
		type = "ipv4-trusthost"
		ipv4_trusthost = "61.149.0.0 255.255.0.0"
	}

	trusthost {
		type = "ipv4-trusthost"
		ipv4_trusthost = "22.22.0.0 255.255.0.0"
	}
	comments = "Terraform Test"
}

```
Thank you for your contribution, the pullrequest will be merged into master in the next release.

dbadola commented:
@frankshen01 , I am using AWS's based Fortigate v6.0.5 for my test and have given superuser access to the API user but it's still failing. Maybe it's related to VM based fortigates

frankshen01 commented:
TESTACC passed

```

make: Entering directory '/samba/combinfordb/src/github.com/terraform-providers/terraform-provider-fortios'
go clean -testcache $(go list ./... |grep -v 'vendor')
make: Leaving directory '/samba/combinfordb/src/github.com/terraform-providers/terraform-provider-fortios'
make: Entering directory '/samba/combinfordb/src/github.com/terraform-providers/terraform-provider-fortios'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test $(go list ./... |grep -v 'vendor') -v  -timeout 120m
?       github.com/terraform-providers/terraform-provider-fortios       [no test files]
=== RUN   TestProvider
--- PASS: TestProvider (0.00s)
=== RUN   TestAccFortiOSFirewallObjectAddress_withIPRange
--- PASS: TestAccFortiOSFirewallObjectAddress_withIPRange (1.46s)
=== RUN   TestAccFortiOSFirewallObjectAddress_withGeography
--- PASS: TestAccFortiOSFirewallObjectAddress_withGeography (1.39s)
=== RUN   TestAccFortiOSFirewallObjectAddress_withFqdn
--- PASS: TestAccFortiOSFirewallObjectAddress_withFqdn (3.19s)
=== RUN   TestAccFortiOSFirewallObjectAddressGroup_basic
--- PASS: TestAccFortiOSFirewallObjectAddressGroup_basic (2.41s)
=== RUN   TestAccFortiOSFirewallObjectIPPool_Overload
--- PASS: TestAccFortiOSFirewallObjectIPPool_Overload (1.04s)
=== RUN   TestAccFortiOSFirewallObjectIPPool_OneToOne
--- PASS: TestAccFortiOSFirewallObjectIPPool_OneToOne (3.95s)
=== RUN   TestAccFortiOSFirewallObjectService_Fqdn
--- PASS: TestAccFortiOSFirewallObjectService_Fqdn (1.40s)
=== RUN   TestAccFortiOSFirewallObjectService_IPRange
--- PASS: TestAccFortiOSFirewallObjectService_IPRange (3.24s)
=== RUN   TestAccFortiOSFirewallObjectServiceGroup_basic
--- PASS: TestAccFortiOSFirewallObjectServiceGroup_basic (2.35s)
=== RUN   TestAccFortiOSFirewallObjectVip_basic
--- PASS: TestAccFortiOSFirewallObjectVip_basic (1.03s)
=== RUN   TestAccFortiOSFirewallObjectVipGroup_basic
--- PASS: TestAccFortiOSFirewallObjectVipGroup_basic (4.99s)
=== RUN   TestAccFortiOSFirewallSecurityPolicy_basic
--- PASS: TestAccFortiOSFirewallSecurityPolicy_basic (3.64s)
=== RUN   TestAccFortiOSLogFortiAnalyzerSetting_basic
--- PASS: TestAccFortiOSLogFortiAnalyzerSetting_basic (1.32s)
=== RUN   TestAccFortiOSLogSyslogSetting_basic
--- PASS: TestAccFortiOSLogSyslogSetting_basic (1.31s)
=== RUN   TestAccFortiOSNetworkingInterfacePort_Physical
--- PASS: TestAccFortiOSNetworkingInterfacePort_Physical (3.30s)
=== RUN   TestAccFortiOSNetworkingInterfacePort_Vlan
--- PASS: TestAccFortiOSNetworkingInterfacePort_Vlan (1.35s)
=== RUN   TestAccFortiOSNetworkingInterfacePort_Loopback
--- PASS: TestAccFortiOSNetworkingInterfacePort_Loopback (1.54s)
=== RUN   TestAccFortiOSNetworkingRouteStatic_basic
--- PASS: TestAccFortiOSNetworkingRouteStatic_basic (3.62s)
=== RUN   TestAccFortiOSSystemAdminAdministrator_basic
--- PASS: TestAccFortiOSSystemAdminAdministrator_basic (2.33s)
=== RUN   TestAccFortiOSSystemAdminProfiles_basic
--- PASS: TestAccFortiOSSystemAdminProfiles_basic (3.04s)
=== RUN   TestAccFortiOSSystemAPIUserSetting_basic
--- PASS: TestAccFortiOSSystemAPIUserSetting_basic (1.40s)
=== RUN   TestAccFortiOSSystemSettingDNS_basic
--- PASS: TestAccFortiOSSystemSettingDNS_basic (1.56s)
=== RUN   TestAccFortiOSSystemSettingGlobal_basic
--- PASS: TestAccFortiOSSystemSettingGlobal_basic (3.04s)
=== RUN   TestAccFortiOSSystemSettingNTP_basic
--- PASS: TestAccFortiOSSystemSettingNTP_basic (1.32s)
=== RUN   TestAccFortiOSVPNIPsecPhase1Interface_basic
--- PASS: TestAccFortiOSVPNIPsecPhase1Interface_basic (1.60s)
=== RUN   TestAccFortiOSVPNIPsecPhase1Interface_basic2
--- PASS: TestAccFortiOSVPNIPsecPhase1Interface_basic2 (3.24s)
=== RUN   TestAccFortiOSVPNIPsecPhase2Interface_basic1
--- PASS: TestAccFortiOSVPNIPsecPhase2Interface_basic1 (4.76s)
=== RUN   TestAccFortiOSVPNIPsecPhase2Interface_basic2
--- PASS: TestAccFortiOSVPNIPsecPhase2Interface_basic2 (1.51s)
PASS
ok      github.com/terraform-providers/terraform-provider-fortios/fortios       66.379s
make: Leaving directory '/samba/combinfordb/src/github.com/terraform-providers/terraform-provider-fortios'

```
